core-rust/state-manager/src/transaction/executable_logic.rs

```diff
@@ -259,3 +275,47 @@ fn collect_new_node_ids(state_updates: &StateUpdates) -> IndexSet<NodeId> {
         .map(|(node_id, _node_state_updates)| *node_id)
         .collect()
 }
+
+/// An extension trait for easier, declarative customization of our various [`ExecutionConfig`]s.
+trait CustomizedExecutionConfig {
+    fn with_costing_parameters(self, costing_parameters: CostingParameters) -> Self;
+    fn with_limit_parameters(self, limit_parameters: LimitParameters) -> Self;
+}
+
+impl CustomizedExecutionConfig for ExecutionConfig {
+    fn with_costing_parameters(self, costing_parameters: CostingParameters) -> Self {
+        let ExecutionConfig {
+            enable_kernel_trace,
+            enable_cost_breakdown,
+            execution_trace,
+            system_overrides,
+        } = self;
+        ExecutionConfig {
+            enable_kernel_trace,
+            enable_cost_breakdown,
+            execution_trace,
+            system_overrides: Some(SystemOverrides {
+                costing_parameters: Some(costing_parameters),
+                ..system_overrides.expect("all ExecutionConfig's constructors set this field")
```

Comment:
Perhaps we should just use `unwrap_or_default()` here? Although arguably upstream we could just get rid of `Some(SystemOverrides)` and force SystemOverrides to be provided